src/subgraphs/getData.ts

```diff
@@ -83,31 +86,50 @@ export class DataRecoveryGraph extends AbstractGraph<DataRecoveryState> {
         value: (x: string, y?: string) => (y ? y : x),
         default: () => '',
       },
+      status: {
+        value: (x: NodeStatus | null, y?: NodeStatus | null) => (y ? y : x),
+        default: () => null,
+      },
     };
     super(graphState);
     this.functions = functions;
     this.database = database;
     this.companyName = companyName;
+    this.outputHandler = outputHandler;
+  }
+
+  private async handleStatusUpdate(status: NodeStatus) {
+    if (status) {
+      if ('error' in status) {
+        this.outputHandler('error', status.error!.message);
+      } else if (status.type = 'data') {
+        this.outputHandler('status', status.data!.message);
+      }
+    }
   }
 
   private async fetchSchemaNode(state: DataRecoveryState): Promise<DataRecoveryState> {
-    const { sessionToken, schema } = await fetchSchema(this.companyName, this.database);
-    return { ...state, sessionToken, schema };
+    const { sessionToken, schema, status } = await fetchSchema(this.companyName, this.database);
```

Comment:
NodeStatus is a property of the nodes of subgraphs, we may want to use fetchSchema in other parts of the code and want a regular error

fetchSchema should throw or return an error (Whatever we do in metabase api)

And in this file we create the status

Reply:
What do you mean? Not sure I understand